ocaml/xapi-idl/storage/storage_interface.ml

```diff
@@ -291,12 +291,39 @@ module Mirror = struct
   }
   [@@deriving rpcty]
 
-  type mirror_receive_result = Vhd_mirror of mirror_receive_result_vhd_t
+  type mirror_receive_result_qcow2_t = {
+      mirror_vdi: vdi_info
+    ; mirror_datapath: dp
+    ; nbd_export: string
+  }
+  [@@deriving rpcty]
+
+  type mirror_receive_result =
+    | Vhd_mirror of mirror_receive_result_vhd_t
+    | QCOW2_mirror of mirror_receive_result_qcow2_t
   [@@deriving rpcty]
 
   type similars = content_id list [@@deriving rpcty]
+
+  type copy_operation_v1 = string [@@deriving rpcty, show {with_path= false}]
+
+  type mirror_operation_v1 = string [@@deriving rpcty, show {with_path= false}]
+
+  (* SMAPIv3 mirror operation *)
+  type operation =
+    | CopyV1 of copy_operation_v1
```

Comment:
Now the v's are getting confusing... But this seems to be defined in xapi-storage?

Reply:
yep